src/components/Toolbox.tsx

```diff
@@ -118,6 +120,37 @@ export default function Toolbox(props: React.PropsWithChildren<ToolboxProps>) {
     setIsUnsavedDialogVisible(false);
   };
 
+  // handler for JSON error messages
+  const [isErrorVisible, setIsErrorVisible] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const showError = (message: string) => {
+    setErrorMessage(message);
+    setIsErrorVisible(true);
+  };
+
+  const handleErrorClose = () => {
+    setIsErrorVisible(false);
+  };
+
+  const handleFileUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    StateManager.uploadJSON(e)
+      .then((parsedData) => {
+        const automatonCheck = StateManager.isValidAutomaton(parsedData);
+        if (automatonCheck[0] == false) {
+          showError(automatonCheck[1]);
+          return;
+        }
+
+        StateManager.loadAutomaton(parsedData);
+      })
+      .catch((response) => {
+        showError(
+          "Input file must contain valid JSON. Download a sample Automaton for an example.",
```

Comment:
```suggestion
          "The file does not contain valid JSON.",
```
At the moment, I don't think we have sample automata for users to download, so we shouldn't be suggesting that they look for any. I think having a sample library would be a good idea - perhaps we could add that as an issue.